citystatmodel.h

```diff
@@ -0,0 +1,62 @@
+#ifndef CITYSTATMODEL_H
+#define CITYSTATMODEL_H
+
+#include "cityitem.h"
+
+#include <QAbstractTableModel>
+#include <QList>
+#include <QString>
+#include <memory>
+#include <limits>
+
+class DataLoader;
+
+class CityStatModel : public QAbstractTableModel
+{
+    Q_OBJECT
+
+public:
+    explicit CityStatModel(QObject *parent = nullptr);
+    ~CityStatModel();
+
+    // Header:
+    QVariant headerData(int section, Qt::Orientation orientation, int role = Qt::DisplayRole) const override;
+    //bool setHeaderData(int section, Qt::Orientation orientation, const QVariant &value, int role = Qt::EditRole) override;
+
+    // Basic functionality:
+    int rowCount(const QModelIndex &parent = QModelIndex()) const override;
+    int columnCount(const QModelIndex &parent = QModelIndex()) const override;
+
+    QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const override;
+
+    // Editable:
+    bool setData(const QModelIndex &index, const QVariant &value,
+                 int role = Qt::EditRole) override;
+
+    Qt::ItemFlags flags(const QModelIndex& index) const override;
+
+    // Add data:
+    bool insertRows(int row, int count, const QModelIndex &parent = QModelIndex()) override;
+    // Remove data:
+    bool removeRows(int row, int count, const QModelIndex &parent = QModelIndex()) override;
+
+    //fill population info
+    void fillPopulationInfo();
+public slots:
+    void saveDataInitiated();
+
+signals:
+    void isSaveDataSuccessfull(bool isSavingSuccessfull, unsigned int savedItemsCounter);
+    void populationChanged(unsigned int maxPopulation,unsigned int minPopulation,unsigned int totalPopulation);
+
+private:
+    QList <CityItem> m_cityItemsList;
+    std::unique_ptr<DataLoader> m_dataLoader;
+
+    unsigned int m_minPopulationElement = std::numeric_limits<unsigned int>::max();
```

Comment:
Every time we edit the existing items or add the new data, we walk through the vector of all items end look for min,max, count total.
We must have an value to compare with. There are 2 ways to do it:
- We set the first item's value as min/max and then compare the next value with the first one.
- When we make the first check, the first item has to be compared with the biggest value of unsigned int. 
We cannot set min value to the first item's value in header, so that is why I have used the second way.